client.go

```diff
@@ -72,6 +74,14 @@ func WithHTTPClient(httpClient *http.Client) ClientOption {
 	}
 }
 
+// WithClientHeaders allows to set custom headers defined at client level, used
+// in all requests.
+func WithClientHeaders(headers map[string]string) ClientOption {
```

Comment:
This could be renamed to `WithHeaders`, as this is going to be used only in the context of creating a new client, the word "client" in the name is redundant.

This feels very natural:
```golang
NewClient(apikey, WithHeaders(...))
```

While this is more verbose than necessary:
```golang
NewClient(apikey, WithClientHeaders(...))
```

Reply:
Also, I'm now noticing that `WithHeader` (singular) already exists. So, for coherency the plural form `WithHeaders` makes more sense. 

But this introduces other questions:

1. Is really necessary to introduce a new option `WithHeaders` when `WithHeader` can be used multiple times for specifying more than one header?

2. Using `WithHeaders` and `WithHeader` at the same time can have unexpected results, which depend in the order in which the calls are made. Calling `WithHeaders` removes any header previously added using `WithHeader`, however, calling `WithHeader` after `WithHeaders` adds/replaces the header.